js/data/bucket.js

```diff
@@ -109,6 +109,35 @@ Bucket.prototype.populateArrays = function() {
 };
 
 /**
+ * Update paint arrays with the given feature properties, leaving geometries
+ * as-is.
+ * @private
+ */
+Bucket.prototype.updatePaintVertexArrays = function(programName, propertiesList) {
+    this.recalculateStyleLayers();
+
+    for (var i = 0; i < propertiesList.length; i++) {
+        var range = this._featureIndexToArrayRange[i];
+        if (!range) continue;
+        var numVertices = range.endVertex - range.startVertex + (range.endGroup - range.startGroup) * ArrayGroup.MAX_VERTEX_ARRAY_LENGTH;
+        this.prepareArrayGroup(programName, numVertices);
```

Comment:
Oh, hold on -- actually, this call to `prepareArrayGroup` is happening once per feature here, which I think makes it quite a bit less likely that we would exceed `ArrayGroup.MAX_VERTEX_ARRAY_LENGTH`.  (Consider, e.g., https://github.com/mapbox/mapbox-gl-js/blob/master/js/data/bucket/fill_bucket.js#L79 -- although it's true that that is on an individual polygon ring rather than the whole feature.)